google-cloud-spanner/src/main/java/com/google/cloud/spanner/DelayedMultiplexedSessionTransaction.java

```diff
@@ -224,4 +225,22 @@ private SessionReference getSessionReference() {
       throw SpannerExceptionFactory.propagateInterrupt(interruptedException);
     }
   }
+
+  /**
+   * Execute `stmt` within PARTITIONED_DML transaction using multiplexed session. This method is a
+   * blocking call as the interface expects to return the output of the `stmt`.
+   */
+  @Override
+  public long executePartitionedUpdate(Statement stmt, UpdateOption... options) {
+    try {
+      SessionReference sessionReference = this.sessionFuture.get();
```

Comment:
Use:

```suggestion
      SessionReference sessionReference = getSessionReference();
```

This automatically handles the exceptions, so you can also get rid of the throw-catch block.